### PR TITLE
Use -H:PrintAnalysisCallTreeType=CSV to generate csv call tree files

### LINKED ISSUE #20733: Default native reports to CSV
### Description

When implemented, integrate https://github.com/oracle/graal/issues/3843 into Quarkus.

When a user sets `-Dquarkus.native.enable-reports`, Quarkus should instruct GraalVM to produce CSV call trees by default.

Quarkus should not offer a specific property for switching to a text format call tree. The user can always pass in `-Dquarkus.native.additional-build-args=+-H:+PrintAnalysisCallTreeType=text` if they really want the text format.

FYI @zakkak @Sanne 

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by zakkak**
Fixes #20733

~Marking this as draft till I learn [whether the parameter will be backported to 21.3 as well or not](https://github.com/oracle/graal/pull/3861#issuecomment-1064307037).~

### CODE REVIEW

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks!

**Comment by zakkak** [comment]
Marked this for backport to 2.7 as the flag is going to land in 21.3.2 as well which 2.7 will be compatible with.

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java [hunk 1]
```diff
@@ -665,6 +665,19 @@ public NativeImageInvokerInfo build() {
                 //address https://github.com/quarkusio/quarkus-quickstarts/issues/993
                 nativeImageArgs.add("-J--add-opens=java.base/java.text=ALL-UNNAMED");
 
+                if (nativeConfig.enableReports) {
+                    if (graalVMVersion.isOlderThan(GraalVM.Version.VERSION_21_3_2)) {
+                        nativeImageArgs.add("-H:+PrintAnalysisCallTree");
+                    } else {
+                        nativeImageArgs.add("-H:PrintAnalysisCallTreeType=CSV");
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java:672]: @zakkak can you confirm that having `-H:PrintAnalysisCallTreeType=CSV` triggers `-H:+PrintAnalysisCallTree` and we don't need to add it explicitly?

Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java:672]: Yes. It happens [here](https://github.com/oracle/graal/blob/59a4294796c260fbadc3e1fee957928d2fbf3d48/substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/reports/AnalysisReportsOptions.java#L43-L50).

**Comment by rsvoboda** [comment]
@zakkak I think `src/main/asciidoc/native-reference.adoc` should be enhanced to mention this is enabled by default and additional-build-args is not needed anymore

https://quarkus.io/guides/native-reference#call-tree-reports

**Comment by zakkak** [comment]
Good point @rsvoboda 

Documentation updated in https://github.com/quarkusio/quarkus/pull/26133

**Pull request merged**